File: 2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.py
class Solution:
    def intersection(self, nums: List[List[int]]) -> List[int]:
        counter = Counter()
        result = []
        n = len(nums)

        for row in nums:
            for num in row:
                counter[num] += 1

        for num, freq in counter.items():
            if freq == n:
                result.append(num)
        
        return result

File: 2248-intersection-of-multiple-arrays/2248-intersection-of-multiple-arrays.java
class Solution {
    public List<Integer> intersection(int[][] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        List<Integer> result  = new ArrayList<>();

        for(int[] row : nums){
            for(int num : row){
               map.put(num, map.getOrDefault(num, 0) + 1);
            }
        }

        for(int key : map.keySet()){
            if(map.get(key) == nums.length){
                result.add(key);
            }
        }
        
        Collections.sort(result);
        return result;
    }
}